### ISSUE TITLE
CPUID with eax = 0 not implemented

### ISSUE THREAD

**Issue by PhilippTakacs** [unicorn-engine/unicorn#1515]
Hi

Calling cpuid with eax = 0 should set eax to Maximum Input Value for Basic CPUID Information. Also ebx, ecx and edx should be set to a vendor specific string. Here a sample test:

```
#define OK(x) {uc_err __err; if ((__err = x)) { fprintf(stderr, "%s", uc_strerror(__err)); assert(false); } }
static void test_cpuid(void)
{
    uc_engine *uc;
    uint32_t eax;
    uint32_t ebx;

    char code[] = { 0x31, 0xC0, 0x0F, 0xA2}; //XOR EAX EAX; CPUID

    OK(uc_open(UC_ARCH_X86, UC_MODE_32, &uc));
    uc_ctl_set_cpu_model(uc, UC_CPU_X86_486);
    OK(uc_mem_map(uc, 0, 4 * 1024, UC_PROT_ALL));
    OK(uc_mem_write(uc, 0, code, sizeof(code) / sizeof(code[0])));
    OK(uc_emu_start(uc, 0, sizeof(code) / sizeof(code[0]), 0, 0));

    /* Read eax after emulation */
    OK(uc_reg_read(uc, UC_X86_REG_EAX, &eax));
    OK(uc_reg_read(uc, UC_X86_REG_EBX, &ebx));

    assert(eax != 0);
    assert(ebx == 0x756e6547); //magic string "Genu" for intel cpu

    OK(uc_close(uc));
}
```

**Comment by wtdcode** [comment]
What's the actual and expected output?

**Comment by PhilippTakacs** [comment]
Currently are eax, ebx, ecx and edx all set to 0.
The expected output is depending on the emulated CPU.
For UC_CPU_X86_486 eax should be at least 1, ebx 0x756e6547, ecx 0x6c65746e and edx 0x49656e69.

**Comment by wtdcode** [comment]
Confirmed it should be a bug. I would look into it asap.

**Comment by wtdcode** [comment]
Fixed in ef6f8a242702f029cdcee298e851f25194bb4070. Thanks.
